src/checkers/inference/InferenceTreeAnnotator.java

```diff
@@ -276,6 +293,25 @@ public Void visitNewClass(final NewClassTree newClassTree, final AnnotatedTypeMi
 
         annotateMethodTypeArgs(newClassTree);
 
+
+        if (newClassTree.getClassBody() != null) {
+            // For a fully annotated anonymous class instantiation as follows,
+            //     new @VarAnnot(1) A() @VarAnnot(2) {...}
+            // create the implied equality constraint "1 == 2"
+            ConstraintManager constraintManager = InferenceMain.getInstance().getConstraintManager();
+            SlotManager slotManager = InferenceMain.getInstance().getSlotManager();
+
+            // Get the varSlot on the type identifier
+            final Slot identifierSlot = slotManager.getSlot(atm);
```

Comment:
Why do you make some local variables final, but not all? I would remove the finals.